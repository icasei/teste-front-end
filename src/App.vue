<template>
  <div id="app">
    <router-view />
  </div>
</template>

<script>
export default {
  name: 'app',
}
</script>

<style>
@import url('https://fonts.googleapis.com/css?family=Arvo:400,700&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  font-family: Arvo, serif;
  font-weight: 400;
  font-size: 16px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #777;
}

a {
  text-decoration: none;
  color: inherit;
}
</style>
