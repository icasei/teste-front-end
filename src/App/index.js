import React from 'react';

import Routes from '../Config/Routes'


function App() {
  return <Routes />
}

export default App;