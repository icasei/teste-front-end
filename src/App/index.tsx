import * as React from 'react'

interface Props {}

const App: React.FC<Props> = () => {
  return (
    <>
      <div>App</div>
    </>
  )
}

export default App
