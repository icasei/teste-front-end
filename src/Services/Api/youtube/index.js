import Fetch from '../../Api'


//TODO need to paste your api_key here
const api_key = ''

export const getYoutubeVideos = (query) =>
  Fetch.get(`/search?part=snippet&key=${api_key}&q=${query}`)