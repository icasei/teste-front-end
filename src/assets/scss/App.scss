@import 'common';

.app {
  width: auto;
  height: 100%;
  margin: auto;
  padding: 1em;
}
