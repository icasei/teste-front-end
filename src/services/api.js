import axios from 'axios'

const API_KEY = 'AIzaSyC1-qtw-x8ocITOYzSElz2LLAZlr7hSsE0'
const API_URL= 'https://www.googleapis.com/youtube/v3/'

export default class {

    constructor() {
        this.apiToken = API_KEY;
        this.apiURL = API_URL;
    }

    async search(searchText, itemsNumber) {
        const endpoint = this.apiURL + 'search?part=id,snippet&q=' + searchText + '&key=' + this.apiToken + '&maxResults=' + itemsNumber + '&type=video';

        const answer = await axios.get(endpoint).then((result) => {
            const apiData = result.data;

            let output = {
                nextPageToken: apiData.nextPageToken,
                pageQuantity: this.getNumberOfPages(apiData),
                results: apiData.items.filter((item) => {
                    item['searchText'] = searchText;
                    return true;
                })
            }
            return output;
        }).catch((error) => {
            return 'error';
        })

        return answer;
    }

    async getPage(page, searchText, itemsNumber) {
        const endpoint = this.apiURL + 'search?part=id,snippet&q=' + searchText + '&key=' + this.apiToken + '&maxResults=' + itemsNumber + '&type=video&pageToken=' + page;

        const answer = await axios.get(endpoint).then((result) => {
            const apiData = result.data;

            let output = {
                nextPageToken: apiData.nextPageToken,
                pageQuantity: this.getNumberOfPages(apiData),
                results: apiData.items
            }
            return output;
        }).catch((error) => {
            return 'error';
        })

        return answer;
    }

    getNumberOfPages(apiData) {
        return Math.round(parseFloat(apiData.pageInfo.totalResults) / parseFloat(apiData.pageInfo.resultsPerPage));
    }

    async getVideoInfo(videoId) {
        const endpoint = this.apiURL + 'videos?id=' + videoId + "&part=snippet,statistics&key=" + this.apiToken;

        return await axios.get(endpoint).then((result) => {
            return result.data
        }).catch((error) => {
            return 'error'
        }
        );
    }

}