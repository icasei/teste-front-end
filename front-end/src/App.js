import React, {  useState, useCallback } from 'react';
import './App.css';
import axios from 'axios';
import debounce from "lodash/debounce";
import{ VideoList, VideoList1, VideoList2 } from './components/VideoList';
import { Box, Button, Container, Grid, Icon, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Videologo from './img/VideoLogo.png'
import { Footer } from './components/Footer';

const useStyles = makeStyles({
  boxSpace: {
    height:'50vh'
  },
  container:{
    textAlign: "center",
    justifyContent: "center",
    '& img': {
      height: 'auto',
      width: 'auto',
      '@media (max-width: 600px)':{
        width: "340px",
        height: "160px",
      }
    },
    '& input': {
      margin: 5,
      padding: 5,
      width: '250px',
      height: '30px',
    }
  },
  hidder:{
    marginTop: '150px',
  }
});


const App = () => {
  const [videos, setVideos] = useState();
  const [more, setMore] = useState(0);
  const KEY = 'AIzaSyDyRShtY1aPx8Atejb65Ip_IknBZwv1JEY';
  const classes = useStyles();
  

  const fetchApi = useCallback(async (search) => {
    try {
      const res = await axios.get('https://www.googleapis.com/youtube/v3/search',
      {
        params: {
          part: 'snippet',
          maxResults: 30,
          key: KEY,
          q: search,
        },
      }
      );
      setMore(0)
      setVideos(res.data);
    } catch (err) {
      console.log(err);
    }
  }, []);

    const searchHandler = debounce(async (event) => {
      const { value } = event.target;
      if (value === "") return setVideos(null);
      await fetchApi(value);
    }, 1000);
    
    function moreVideos() {
      setMore(more => more + 1)
    }

  return (
    <>
    <Container 
    className={classes.container}
    justify="center"
    alignItems="center"
    >
      <Grid>
        <img src={Videologo} />
      </Grid>
      <Grid item className={ !videos? classes.hidder : ''} xs={12} >
        <input
          type="text"
          placeholder="pesquisa..."
          onChange={searchHandler}
        />
      </Grid>

      {!videos ? 
      <Box className={classes.boxSpace}>
        <Typography variant="h1">bem-vindo</Typography>
      </Box>
      :
      <>
      <Grid item lg={12}>
        <Box component="div" display={more >=0? '': 'none'} >
            <VideoList videos={videos.items} /> 
        </Box>
        <Box component="div" display={more >=1? '': 'none'} >
          <VideoList1  videos={videos.items} /> 
        </Box>
        <Box component="div" display={more >=2? '': 'none'} >
          <VideoList2 videos={videos.items} />
        </Box>
        </Grid>
        <Grid item lg={12}>
          <Button 
          variant="contained" 
          color="primary" 
          size="large"  
          endIcon={<Icon>keyboard_arrow_down</Icon>}
          disabled={more >= 2 ? true : false}  
          onClick={moreVideos} 
          >
            Mais Videos
          </Button>
        </Grid>
      </>
    }
    </Container>
    <Footer />
    </>
  );
}

export default App;
