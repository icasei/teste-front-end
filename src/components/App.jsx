import 'assets/scss/App.scss'
import React, {Component} from 'react'
import {getVideos} from '../youtube/api'
import Search from 'components/Search'
import VideoList from 'components/VideoList'
import Pagination from 'components/Pagination'
import Spinner from 'components/Spinner'
import VideoDetail from 'components/VideoDetail'
import Carousel from 'components/carousel/Carousel'


export default class App extends Component {
  constructor() {
    super()
    this.state = {}
  }
  
  scrollUp() {
    var doc = document.documentElement
    var top = (window.pageYOffset || doc.scrollTop) - (doc.clientTop || 0)

    if (top > 0) {
      // window.scrollTo(0, top - 15)
      window.scrollTo(0, 0)
    }
  }

  onSearch(query, page = '') {
    this.setState({
        query: query,
        queryResult: undefined,
        prevPageToken: undefined,
        nextPageToken: undefined
      }, 
      () => {
        getVideos(query, data => {
          this.setState({queryResult: data.data})
        }, page)

        if (page.length > 0) {
          this.scrollUp()
        }
      }
    )
  }

  onPagination(prev, next) {
    this.setState({prevPageToken: prev, nextPageToken: next})
  }

  showDetail(data) {
    this.setState({videoDetail: data})
    this.scrollUp()
  }

  hideDetail() {
    this.setState({videoDetail: undefined})
  }

  renderResults() {
    if (this.state.videoDetail) {
      return (
        <VideoDetail data={this.state.videoDetail} hideDetail={this.hideDetail.bind(this)} />
      )
    } else {
      return (
        <div>
          {this.state.queryResult
            ? 
              <div>
                <VideoList data={this.state.queryResult} onPagination={this.onPagination.bind(this)} showDetail={this.showDetail.bind(this)} />
                <Carousel data={this.state.queryResult} showDetail={this.showDetail.bind(this)} xsQuantity={12} smQuantity={6} mdQuantity={3} speed={5} />
              </div>
            : this.state.query && <Spinner />
          }

          {(this.state.prevPageToken || this.state.nextPageToken) &&
            <Pagination onSearch={this.onSearch.bind(this)} query={this.state.query} prev={this.state.prevPageToken} next={this.state.nextPageToken} />
          }
        </div>
      )
    }
  }

  render() {
    return (
      <div className="app">
        <Search onSearch={this.onSearch.bind(this)} />
        {this.renderResults()}
      </div>
    )
  }
}