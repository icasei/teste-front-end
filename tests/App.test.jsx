import React from 'react'
import renderer from 'react-test-renderer'
import App from '../src/components/App'
import {getVideos, videoDetail} from '../src/youtube/api'


jest.mock('react-dom');


it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot()
})

it('youtube get videos', () => {

})

it('youtube get video detail', () => {

})