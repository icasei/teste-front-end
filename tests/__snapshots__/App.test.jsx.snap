// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18 search"
  >
    <div
      className="logo"
    >
      <img
        height="100"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg=="
      />
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-8-37"
    >
      <div
        className="MuiInput-root-98 MuiInput-error-103 MuiInput-fullWidth-105 MuiInput-underline-102"
        onKeyPress={[Function]}
        required={true}
      >
        <input
          aria-invalid={true}
          aria-label="Buscar"
          autoFocus={true}
          className="MuiInput-input-106"
          id="searchInput"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Pesquisar"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
    >
      <button
        className="MuiButtonBase-root-117 MuiIconButton-root-111"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-116"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-120"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-127"
        />
      </button>
    </div>
  </div>
  <div />
</div>
`;
